python/setup.py

```diff
@@ -30,7 +30,12 @@
 else:
     from setuptools import setup
     from setuptools.extension import Extension
-    kwargs = {'install_requires': ['numpy>1.16.0,<2.0.0', 'requests>=2.20.0,<3', 'graphviz<0.9.0,>=0.8.1'], 'zip_safe': False}
+    kwargs = {
+        'install_requires': ['requests>=2.20.0,<3', 'graphviz<0.9.0,>=0.8.1']
+            .append('numpy>=1.17.0,<2.0.0' if sys.version_info.major >= 3 and sys.version_info.minor >= 5
+                    else 'numpy>1.16.0,<2.0.0'),
```

Comment:
what's the concern for just switching to `>= 1.17`? was it CI?
setup.py may be evaluated on a platform that builds binary distributions that is different from the platform on which the distribution is installed. in this scenario, the condition won't be evaluated on the target platform

Reply:
`1.17` is Python 3 only.
If the condition is not evaluated on the target platform, we can't bump the version without dropping Python 2 support..

Reply:
agreed. let's put this change on hold